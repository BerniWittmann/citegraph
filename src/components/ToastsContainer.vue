<template>
  <v-slide-y-transition group class="toast-container">
    <toast v-for="toast in toasts" :key="toast.id" :toast="toast"></toast>
  </v-slide-y-transition>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Toast from '@/models/toast'
import ToastComponent from '@/components/Toast.vue'

@Component({
  components: { Toast: ToastComponent }
})
export default class ToastsContainer extends Vue {
  get toasts (): Array<Toast> {
    return this.$store.getters['toasts/toasts']
  }
}
</script>

<style lang="scss">
.toast-container {
  position: absolute;
  top: 5px;
  right: 10px;
}
</style>
