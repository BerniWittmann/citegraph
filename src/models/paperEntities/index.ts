import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
