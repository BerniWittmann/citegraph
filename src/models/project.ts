export default class Project {
  name: string;
  id?: number;

  constructor ({ name, id }: Project) {
    this.id = id
    this.name = name
  }
}
