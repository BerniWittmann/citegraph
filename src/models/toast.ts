import uuid from 'uuid'

export enum ToastSeverity {
  SUCCESS = 'success',
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error'
}

export enum ToastIcon {
  SUCCESS = 'mdi-check-circle-outline',
  INFO = 'mdi-information-outline',
  WARNING = 'mdi-alert',
  ERROR = 'mdi-alert-circle-outline',
}

export enum ToastDelay {
  SUCCESS = 3000,
  INFO = 4000,
  WARNING = 5000,
  ERROR = 6000,
}

interface ToastData {
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number
}

export default class Toast implements ToastData {
  id: string;
  timestamp: number;
  severity: ToastSeverity;
  icon: ToastIcon;
  text: string;
  delay: number

  constructor ({ severity, icon, text, delay }: ToastData) {
    this.id = uuid.v4()
    this.timestamp = Date.now()
    this.severity = severity
    this.icon = icon
    this.text = text
    this.delay = delay
  }
}

export class SuccessToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.SUCCESS,
      icon: ToastIcon.SUCCESS,
      delay: ToastDelay.SUCCESS
    })
  }
}

export class InfoToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.INFO,
      icon: ToastIcon.INFO,
      delay: ToastDelay.INFO
    })
  }
}

export class WarningToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.WARNING,
      icon: ToastIcon.WARNING,
      delay: ToastDelay.WARNING
    })
  }
}

export class ErrorToast extends Toast {
  constructor (text: string) {
    super({
      text,
      severity: ToastSeverity.ERROR,
      icon: ToastIcon.ERROR,
      delay: ToastDelay.ERROR
    })
  }
}
