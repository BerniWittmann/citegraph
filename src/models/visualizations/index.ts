import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
