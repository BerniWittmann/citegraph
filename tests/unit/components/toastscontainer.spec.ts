import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import ToastsContainer from '@/components/ToastsContainer.vue'
import { ErrorToast, SuccessToast } from '@/models/toast'
import ToastComponent from '@/components/Toast.vue'

describe('components/ToastsContaienr.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(ToastsContainer, {
      i18n,
      mocks: {
        $store: {
          getters: {
            'toasts/toasts': [
              new SuccessToast('Test Message'),
              new ErrorToast('This is an error')
            ]
          }
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the toasts', () => {
    const wrapper = shallowMount(ToastsContainer, {
      i18n,
      mocks: {
        $store: {
          getters: {
            'toasts/toasts': [
              new SuccessToast('Test Message'),
              new ErrorToast('This is an error')
            ]
          }
        }
      }
    })
    const toasts = wrapper.findAll(ToastComponent)
    expect(toasts.length).toEqual(2)
  })

  it('handles an empty array of toasts', () => {
    const wrapper = shallowMount(ToastsContainer, {
      i18n,
      mocks: {
        $store: {
          getters: {
            'toasts/toasts': []
          }
        }
      }
    })
    const toasts = wrapper.findAll(ToastComponent)
    expect(toasts.length).toEqual(0)
  })
})
