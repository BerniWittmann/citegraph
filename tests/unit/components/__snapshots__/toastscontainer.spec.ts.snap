// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ToastsContaienr.vue renders 1`] = `
<v-slide-y-transition-stub group="true" origin="top center 0" class="toast-container">
  <toast-stub toast="[object Object]"></toast-stub>
  <toast-stub toast="[object Object]"></toast-stub>
</v-slide-y-transition-stub>
`;
